extensions-contrib/rocketmq-indexing-service/src/main/java/org/apache/druid/indexing/rocketmq/RocketMQIndexTaskIOConfig.java

```diff
@@ -0,0 +1,174 @@
+      @JsonProperty("endSequenceNumbers")
+      @Nullable SeekableStreamEndSequenceNumbers<String, Long> endSequenceNumbers,
+      @JsonProperty("consumerProperties") Map<String, Object> consumerProperties,
+      @JsonProperty("pollTimeout") Long pollTimeout,
+      @JsonProperty("useTransaction") Boolean useTransaction,
+      @JsonProperty("minimumMessageTime") DateTime minimumMessageTime,
+      @JsonProperty("maximumMessageTime") DateTime maximumMessageTime,
+      @JsonProperty("inputFormat") @Nullable InputFormat inputFormat
+  )
+  {
+    super(
+        taskGroupId,
+        baseSequenceName,
+        startSequenceNumbers == null
+        ? Preconditions.checkNotNull(startPartitions, "startPartitions").asStartPartitions(true)
+        : startSequenceNumbers,
+        endSequenceNumbers == null ? endPartitions : endSequenceNumbers,
+        useTransaction,
+        minimumMessageTime,
+        maximumMessageTime,
+        inputFormat
+    );
+
+    this.consumerProperties = Preconditions.checkNotNull(consumerProperties, "consumerProperties");
+    this.pollTimeout = pollTimeout != null ? pollTimeout : RocketMQSupervisorIOConfig.DEFAULT_POLL_TIMEOUT_MILLIS;
+
+    final SeekableStreamEndSequenceNumbers<String, Long> myEndSequenceNumbers = getEndSequenceNumbers();
+    for (String partition : myEndSequenceNumbers.getPartitionSequenceNumberMap().keySet()) {
+      Preconditions.checkArgument(
+          myEndSequenceNumbers.getPartitionSequenceNumberMap()
+                       .get(partition)
+                       .compareTo(getStartSequenceNumbers().getPartitionSequenceNumberMap().get(partition)) >= 0,
+          "end offset must be >= start offset for partition[%s]",
+          partition
+      );
+    }
+  }
+
+  public RocketMQIndexTaskIOConfig(
+      int taskGroupId,
+      String baseSequenceName,
+      SeekableStreamStartSequenceNumbers<String, Long> startSequenceNumbers,
+      SeekableStreamEndSequenceNumbers<String, Long> endSequenceNumbers,
+      Map<String, Object> consumerProperties,
+      Long pollTimeout,
+      Boolean useTransaction,
+      DateTime minimumMessageTime,
+      DateTime maximumMessageTime,
+      InputFormat inputFormat
+  )
+  {
+    this(
+        taskGroupId,
+        baseSequenceName,
+        null,
+        null,
+        startSequenceNumbers,
+        endSequenceNumbers,
+        consumerProperties,
+        pollTimeout,
+        useTransaction,
+        minimumMessageTime,
+        maximumMessageTime,
+        inputFormat
+    );
+  }
+
+  /**
+   * This method is for compatibilty so that newer version of RocketMQIndexTaskIOConfig can be read by
+   * old version of Druid. Note that this method returns end sequence numbers instead of start. This is because
+   * {@link SeekableStreamStartSequenceNumbers} didn't exist before.
+   */
+  @JsonProperty
+  @Deprecated
+  public SeekableStreamEndSequenceNumbers<String, Long> getStartPartitions()
```

Comment:
As commented above, I think there's no need to provide this method

Reply:
Deleted it.